src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -97,33 +98,63 @@ public Boolean upsertRelation(Node nodeFrom, Node nodeTo, Map<String, String> re
+                    if (recordSize == 0) { // nodes relation doesn't exists
+                        createRelationshipBetweenTwoNodes(nodeFrom, nodeTo, transaction, parameters);
+                    }
+                    else
+                    {
+                        updateRelationshipBetweenTwoNodes("Nodes exists, Updating Relationship Properties!", nodeTo, nodeFrom, statement, result, transaction, recordSize, relationProperties);
+                    }
+                    transaction.commitAsync().toCompletableFuture().get();
+                    logger.info("user relation with toUUID {} and fromUUID {} added successfully ", nodeTo.getId(),
+                            nodeFrom.getId());
                 }
-                transaction.commitAsync().toCompletableFuture().get();
-                logger.info("user relation with toUUID {} and fromUUID {} updated successfully ", nodeTo.getId(),
-                        nodeFrom.getId());
-
-            } catch (ClientException e) {
-                throw new GraphException(ErrorCode.GRAPH_TRANSACTIONAL_ERROR.name(), e.getMessage());
+            } catch (Exception e) {
+                transaction.rollbackAsync().toCompletableFuture().get();
+                logger.error("user relation creation failed : ", e);
+                return Boolean.FALSE;
 
+            } finally {
+                try {
+                    if (session != null && transaction != null) {
+                        transaction.close();
+                        session.close();
+                    }
+                } catch (Exception e) {
+                    logger.error("Session Closed : ", e);
+                }
             }
-        } catch (SessionExpiredException se) {
-            throw new GraphException(ErrorCode.GRAPH_SESSION_EXPIRED_ERROR.name(), se.getMessage());
-        }
-        return true;
+        return Boolean.FALSE;
+    }
+
+    private void updateRelationshipBetweenTwoNodes(String s, Node nodeTo, Node nodeFrom, Statement statement, StatementResult result, Transaction transaction, int recordSize, Map<String, String> relationProperties) {
+        logger.info(s, nodeTo.getId(),
+                nodeFrom.getId());
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("fromUUID", nodeFrom.getId());
+        parameters.put("toUUID", nodeTo.getId());
+        parameters.put(Constants.Graph.PROPS.getValue(), relationProperties);
+        String updateQuery = "MATCH (n:" + label + ")-[r:connect]->(n1:" + label +
+                ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "SET r" + " += " +
+                "$props " + "RETURN n,n1";
+
+        statement = new Statement(updateQuery, parameters);
+        result = transaction.run(statement);
+        recordSize = result.list().size();
+        result.consume();
     }
 
     private void createRelationshipBetweenTwoNodes(Node nodeFrom, Node nodeTo, Transaction transaction, Map<String, Object> parameters) {
```

Comment:
By not returning anything -- we don't know the operation against DB is success or failure... 
can we change this make sure the return value from this used in upsertRelation method properly